mbed_lstools/lstools_darwin.py

```diff
@@ -26,134 +26,86 @@
+        self.mac_version = float('.'.join(platform.mac_ver()[0].split('.')[:2]))
 
+    def find_candidates(self):
         # {volume_id: {serial:, vendor_id:, product_id:, tty:}}
-        volumes = self.get_mbed_volumes()
+        volumes = self.get_volumes()
 
         # {volume_id: mount_point}
         mounts = self.get_mount_points()
-
-        volumes_keys = set(volumes.keys())
-        mounts_keys = set(mounts.keys())
-        intersection = volumes_keys & mounts_keys
-
-        valid_volumes = {}
-
-        for key in intersection:
-            valid_volumes[key] = volumes[key]
-
-        # put together all of that info into the expected format:
-        result =  [
+        return [
             {
                 'mount_point': mounts[v],
                 'serial_port': volumes[v]['tty'],
-                  'target_id': self.target_id(volumes[v]),
-              'platform_name': self.platform_name(self.target_id(volumes[v]))
-            } for v in valid_volumes
+                  'target_id': volumes[v].get('serial')
+            } for v in set(volumes.keys()) & set(mounts.keys())
         ]
 
-        self.ERRORLEVEL_FLAG = 0
-
-        # if we're missing any platform names, try to fill those in by reading
-        # mbed.htm:
-        for i, _ in enumerate(result):
-            if None in result[i]:
-                self.ERRORLEVEL_FLAG = -1
-                continue
-
-            if result[i]['mount_point']:
-                # Deducing mbed-enabled TargetID based on available targetID definition DB.
-                # If TargetID from USBID is not recognized we will try to check URL in mbed.htm
-                htm_target_id = self.get_mbed_htm_target_id(result[i]['mount_point'])
-                if htm_target_id:
-                    result[i]['target_id_usb_id'] = result[i]['target_id']
-                    result[i]['target_id'] = htm_target_id
-                    result[i]['platform_name'] = self.platform_name(htm_target_id[:4])
-                result[i]['target_id_mbed_htm'] = htm_target_id
-
-        return result
-
-
     def get_mount_points(self):
         ''' Returns map {volume_id: mount_point} '''
-
-        # list disks, this gives us disk name, and volume name + mount point:
         diskutil_ls = subprocess.Popen(['diskutil', 'list', '-plist'], stdout=subprocess.PIPE)
-        disks = plistlib.readPlist(diskutil_ls.stdout)
         diskutil_ls.wait()
+        disks = plistlib.readPlist(diskutil_ls.stdout)
 
-        r = {}
-
-        for disk in disks['AllDisksAndPartitions']:
-            mount_point = None
-            if 'MountPoint' in disk:
-                mount_point = disk['MountPoint']
-            r[disk['DeviceIdentifier']] = mount_point
-
-        return r
-
+        return {disk['DeviceIdentifier']: disk.get('MountPoint', None)
+                for disk in disks['AllDisksAndPartitions']}
 
-    def get_mbed_volumes(self):
+    def get_volumes(self):
```

Comment:
I noticed in the linux class, you moved all functions that weren't `find_candidates` to have the private `_` prefix. Do you want to something similar here?

Reply:
Yeah. I also dropped the superfluous `get` prefixes. When everything gets, nothing gets.